internal/events/events_api.go

```diff
@@ -47,8 +44,27 @@ func (a *Api) ListEvents(ctx context.Context, params events.ListEventsParams) mi
 			Severity:  ev.Severity,
 			EventTime: ev.EventTime,
 			Message:   ev.Message,
+			Props:     ev.Props,
 		}
 	}
 	return events.NewListEventsOK().WithPayload(ret)
+}
+
+func CategoryFilter(queryParam []string) []string {
+	result := make([]string, 0)
+
+	if queryParam == nil {
+		//when no categories are requested, proceed with the default behaviour
+		return result
+	}
+
+	if len(queryParam) == 1 && queryParam[0] == "all" {
```

Comment:
It's weird but if a user passes "metrics,all" they would expect it to return all right?

Reply:
In this case the user will get an internal error because once the user passes a list it should be a list of valid categories names. Since now categories are not proliferated as "severity" once were I can scrap the "all" option if you like